### PR TITLE
Add Node 12 support

### PR DESCRIPTION
**Pull request by DavidVujic**
## Description
This PR will make it possible to build this project in Node.js 12. Both deprecated and removed V8 features have been replaced with new equivalent features. Deprecated Nan features have also been replaced. Besides being able to build in Node.js 12, all build warnings about deprecated functions are now fixed.

## Motivation and Context

Node.js 12 will probably very soon be the next LTS. The changes in this PR are critical to make it possible for Node.js 12 projects using this library.

Fixes issue #153 

## How Has This Been Tested?

* Built in Travis CI
* unit tests
* integration tests
* manual testing, using the code in the `examples` folder
* added new component tests: testing new C++ code (`converters.h`)

travis

`npm test`
```bash
total:     108
passing:   108
duration:  12ms
```
`npm run integrationtest`
(verbose output, skipped pasting here)

`npm run test-components`
```bash
total:     8
passing:   8
duration:  8ms
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have read the [code of conduct](https://github.com/yfinkelstein/node-zookeeper/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [contributing guide](https://github.com/yfinkelstein/node-zookeeper/blob/master/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly (if applicable).

### CODE REVIEW

**Comment by DavidVujic** [comment]
I would very much appreciate to get feedback on this PR. My C++ skills are not good, and I am not sure about the new code written follow guidelines or handles memory issues correctly.

Maybe @kuebk can have a look?

Also @jbienkowski311 ?

File: src/converters.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#include <v8.h>
+#include "nan.h"
+
+using namespace v8;
+
+Local<Object> toLocalObj(Local<Value> val) {
```
Comment (jbienkowski311) [inline src/converters.h]: 1) As it's C++ code the file should have `.hpp` extension.
2) We could hide these helper functions behind some kind of namespace.
```cpp
namespace nodezk {
  Local<Object> toLocalObj(Local<Value> val) {
    // ...
  }
}
```

File: src/node-zk.cpp [hunk 2]
```diff
@@ -25,6 +25,7 @@ using namespace node;
 #include "nan.h"
 #include "zk_log.h"
 #include "buffer_compat.h"
+#include "converters.h"
```
Comment (jbienkowski311) [inline src/node-zk.cpp]: `#include "converters.hpp"`

File: src/converters.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    Local<Value> val_local = toLocalVal(arg, propertyName);
+
+    return toBool(val_local);
+}
+
+Local<Value> convertUnixTimeToDate(double time) {
+    return Date::New(Isolate::GetCurrent()->GetCurrentContext(), time).ToLocalChecked();
+}
+
+Local<String> toStr(Local<Value> val) {
```
Comment (jbienkowski311) [inline src/converters.h]: (nitpicking) `toString`?

Comment (DavidVujic) [reply src/converters.h]: I changed the function name to `toString` - was that what you meant? 😄

Comment (jbienkowski311) [reply src/converters.h]: Yes, correct :smile:

**Pull request merged**